pkg/ignition/ignition.go

```diff
@@ -132,3 +166,178 @@ func parseKubeletArgs(unitContents string) (map[string]string, error) {
 	}
 	return kubeletArgs, nil
 }
+
+// GetBootstrapFiles returns the paths of all the files needed to bootstrap kubelet
+func (ign *Ignition) GetBootstrapFiles() []string {
+	files := []string{payload.BootstrapKubeconfigPath, payload.KubeletConfigPath, kubeletCACertPath}
+	for _, ignFile := range ign.config.Storage.Files {
+		if ignFile.Path == cloudConfigPath {
+			files = append(files, cloudConfigPath)
+		}
+	}
+	return files
+}
+
+// CreateBootstrapFiles creates all prerequisite files required to start kubelet in the WMCO payload. Returns file paths
+func (ign *Ignition) CreateBootstrapFiles(c client.Client, apiServerURL, clusterServiceCIDR string) ([]string, error) {
+	// Ensure generated dir exists
+	if _, err := os.Stat(payload.GeneratedDir); errors.Is(err, os.ErrNotExist) {
+		err := os.Mkdir(payload.GeneratedDir, os.ModePerm)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	bootstrapFiles, err := ign.createFilesFromIgnition()
+	if err != nil {
+		return nil, err
+	}
+
+	bootstrapKubeconfigPath, err := createBootstrapKubeconfig(c, apiServerURL)
+	if err != nil {
+		return nil, err
+	}
+	bootstrapFiles = append(bootstrapFiles, bootstrapKubeconfigPath)
+
+	clusterDNS, err := cluster.GetDNS(clusterServiceCIDR)
+	if err != nil {
+		return nil, err
+	}
+	initKubeletConfigPath, err := createKubeletConf(clusterDNS)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I'm a bit concerned about this direction, the files within the payload should be static during the lifetime of the operator, but the files within this function, especially the bootstrap kubeconfig need to be current, using the secret as it is at the time of node bootstrap.

Is this an effort to save cycles? unless theres something I'm missing here, I've brought my concerns about this before. This is a change in behavior that has the potential to prevent nodes from being bootstrapped.